crates/re_build_search_index/src/ingest/python.rs

```diff
@@ -0,0 +1,359 @@
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::process::Command;
+
+use anyhow::Context as _;
+use serde::Deserialize;
+
+use crate::ingest::DocumentData;
+use crate::ingest::DocumentKind;
+use crate::util::CommandExt as _;
+
+use super::Context;
+
+const RERUN_SDK: &str = "rerun_sdk";
+
+pub fn ingest(ctx: &Context) -> anyhow::Result<()> {
+    let progress = ctx.progress_bar("python");
+
+    progress.set_message("mkdocs build");
+    progress.suspend(|| {
+        Command::new("mkdocs")
+            .with_arg("build")
+            .with_arg("-f")
+            .with_arg(ctx.workspace_root().join("rerun_py/mkdocs.yml"))
+            .run_async()
+    })?;
+
+    progress.set_message("sphobjinv convert");
+    let inv: Inventory = Command::new("sphobjinv")
+        .with_args(["convert", "json", "--expand"])
+        .with_cwd(ctx.workspace_root())
+        .with_arg("rerun_py/site/objects.inv")
+        .with_arg("-")
+        .run_serde::<SphinxObjectInv>()
+        .context("sphobjinv may not be installed, install rerun_py/requirements-doc.txt")?
+        .objects
+        .into_values()
+        .map(|o| (o.name.clone(), o))
+        .collect();
+
+    progress.set_message("griffe dump");
+    let dump: Dump = Command::new("griffe")
+        .with_args(["dump", "rerun_sdk"])
+        .run_serde()
+        .context("either griffe or rerun_sdk is not installed")?;
+
+    let docs = collect_docstrings(&dump[RERUN_SDK]);
```

Comment:
it would be nice with comments explaining what all these steps do for those who don't know what `mkdocs`, `sphobjinv`, and `griffe` does